modules/stnBidAdapter.js

```diff
@@ -0,0 +1,488 @@
+          height: adUnit.height,
+          ttl: adUnit.ttl || TTL,
+          creativeId: adUnit.requestId,
+          netRevenue: adUnit.netRevenue || true,
+          nurl: adUnit.nurl,
+          mediaType: adUnit.mediaType,
+          meta: {
+            mediaType: adUnit.mediaType
+          }
+        };
+
+        if (adUnit.mediaType === VIDEO) {
+          bidResponse.vastXml = adUnit.vastXml;
+        } else if (adUnit.mediaType === BANNER) {
+          bidResponse.ad = adUnit.ad;
+        }
+
+        if (adUnit.adomain && adUnit.adomain.length) {
+          bidResponse.meta.advertiserDomains = adUnit.adomain;
+        }
+
+        bidResponses.push(bidResponse);
+      });
+    }
+
+    return bidResponses;
+  },
+  getUserSyncs: function (syncOptions, serverResponses) {
+    const syncs = [];
+    for (const response of serverResponses) {
+      if (syncOptions.iframeEnabled && response.body.params.userSyncURL) {
+        syncs.push({
+          type: 'iframe',
+          url: response.body.params.userSyncURL
+        });
+      }
+      if (syncOptions.pixelEnabled && isArray(response.body.params.userSyncPixels)) {
+        const pixels = response.body.params.userSyncPixels.map(pixel => {
+          return {
+            type: 'image',
+            url: pixel
+          }
+        })
+        syncs.push(...pixels)
+      }
+    }
+    return syncs;
+  },
+  onBidWon: function (bid) {
+    if (bid == null) {
+      return;
+    }
+
+    logInfo('onBidWon:', bid);
+    if (bid.hasOwnProperty('nurl') && bid.nurl.length > 0) {
+      triggerPixel(bid.nurl);
+    }
+  }
+};
+
+registerBidder(spec);
+
+/**
+ * Get floor price
+ * @param bid {bid}
+ * @param mediaType {String}
+ * @param currency {String}
+ * @returns {Number}
+ */
+function getFloor(bid, mediaType, currency) {
+  if (!isFn(bid.getFloor)) {
+    return 0;
+  }
+  let floorResult = bid.getFloor({
+    currency: currency,
+    mediaType: mediaType,
+    size: '*'
+  });
+  return floorResult.currency === currency && floorResult.floor ? floorResult.floor : 0;
+}
+
+/**
+ * Get the ad sizes array from the bid
+ * @param bid {bid}
+ * @param mediaType {String}
+ * @returns {Array}
+ */
+function getSizesArray(bid, mediaType) {
+  let sizesArray = []
+
+  if (deepAccess(bid, `mediaTypes.${mediaType}.sizes`)) {
+    sizesArray = bid.mediaTypes[mediaType].sizes;
+  } else if (Array.isArray(bid.sizes) && bid.sizes.length > 0) {
+    sizesArray = bid.sizes;
+  }
+
+  return sizesArray;
+}
+
+/**
+ * Get schain string value
+ * @param schainObject {Object}
+ * @returns {string}
+ */
+function getSupplyChain(schainObject) {
+  if (isEmpty(schainObject)) {
+    return '';
+  }
+  let scStr = `${schainObject.ver},${schainObject.complete}`;
+  schainObject.nodes.forEach((node) => {
+    scStr += '!';
+    scStr += `${getEncodedValIfNotEmpty(node.asi)},`;
+    scStr += `${getEncodedValIfNotEmpty(node.sid)},`;
+    scStr += `${node.hp ? encodeURIComponent(node.hp) : ''},`;
```

Comment:
Should `node.hp` also use `getEncodedValIfNotEmpty`?

Reply:
Thanks for noticing, updated it!